services/storage/src/simcore_service_storage/simcore_s3_dsm.py

```diff
@@ -549,6 +553,17 @@ async def deep_copy_project_simcore_s3(
         # ensure the full size is reported
         s3_transfered_data_cb.finalize_transfer()
 
+    async def _get_size(self, fmd: FileMetaDataAtDB) -> ByteSize:
+        if not fmd.is_directory:
+            return fmd.file_size
+
+        # in case of directory list files and return size
+        files_meta_data: list[S3MetaData] = await get_s3_client(self.app).list_files(
```

Comment:
is this not huge? should you not use your generator here?

Reply:
🤦 I did all the refactoring to do so here! Thanks for pointing it out

Reply:
I had a feeling I was missing something